visual_qa/florence-2/README.md

```diff
@@ -0,0 +1,178 @@
+- `<DENSE_REGION_CAPTION>`: Dense Region Captioning
+- `<REGION_PROPOSAL>`: Region Proposal
+- `<OCR>`: Optical Character Recognition
+- `<OCR_WITH_BOXES>`: OCR with Bounding Boxes
+- `<REGION_TO_SEGMENTATION>`: Segment a Specific Region
+- `<REGION_TO_CATEGORY>`: Categorize a Specific Region with a One Word Descriptor
+- `<REGION_TO_DESCRIPTION>`: Describe a Specific Region
+- `<REFERRING_EXPRESSION_SEGMENTATION>`: Caption to Generating a Segment
+- `<OPEN_VOCABULARY_DETECTION>`: OCR and Object Detection
+
+For more information, refer [here](https://huggingface.co/microsoft/Florence-2-large/blob/main/sample_inference.ipynb).
+
+### Text Input
+
+For certain tasks, like `<CAPTION_TO_PHRASE_GROUNDING>`, you can supply a prompt to the `text_input` parameter to focus on detecting/segmenting particular objects. Good phrases are 1-2 words long, as adjectives can be picked up as objects, nouns, and verbs.
+
+For more information, refer [here](https://huggingface.co/microsoft/Florence-2-large/blob/main/sample_inference.ipynb).
+
+## Examples
+
+![Example Image](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/car.jpg?download=true)
+
+Let's apply Sieve's Florence-2 to this photo of a car! Be sure to log in via `sieve login` on your terminal or by setting the `SIEVE_API_KEY` environment variable.
+
+### Object Detection
+
+For object detection, use a code snippet like this. 
+
+```python
+
+import sieve
+
+fl2_fn = sieve.function.get("sieve/florence-2")
+
+debug_image, response = fl2_fn.run(
+    image=sieve.File("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/car.jpg?download=true"),
+    task_prompt="<OD>",
+    debug_visualization=True,
+)
+
+print("debug image path:", debug_image.path)
+print("dict response:", response)
+```
+
+If you run on the image of the car, the response dictionary will look something like:
+
+```json
+{
+  "<OD>": {
+    "bboxes": [
+      [...],  // abbreviated for brevity
+      [...],
+      [...],
+      [...]
+    ],
+    "labels": [
+      "car",
+      "door",
+      "wheel",
+      "wheel"
+    ]
+  }
+}
+```
+
+### Guided Object Detection with Prompts
+
+To detect specific objects using text, use a code snippet like this.
+
+```python
+
+import sieve
+
+fl2_fn = sieve.function.get("sieve/florence-2")
+
+text_input = "car, door."
+
+debug_image, response = fl2_fn.run(
+    image=sieve.File("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/car.jpg?download=true"),
+    task_prompt="<CAPTION_TO_PHRASE_GROUNDING>",
+    text_input=text_input,
+    debug_visualization=True,
+)
+
+print("debug image path:", debug_image.path)
+print("dict response:", response)
+```
+
+If you run on the image of the car, the response dictionary will look something like:
+
+```json
+{
+  "<CAPTION_TO_PHRASE_GROUNDING>": {
+    "bboxes": [
+      [...],  // abbreviated for brevity
```

Comment:
u can keep this but "abbreviated for brevity" is one of the most redundant 3-word combinations you could possibly assemble